lib/fs/debug.go

```diff
@@ -18,5 +18,11 @@ var (
 )
 
 func init() {
-	l.SetDebug("fs", strings.Contains(os.Getenv("STTRACE"), "fs") || os.Getenv("STTRACE") == "all")
+	logger.DefaultLogger.NewFacility("walkfs", "Filesystem access while walking")
+	if strings.Contains(os.Getenv("STTRACE"), "walkfs") || os.Getenv("STTRACE") == "all" {
+		l.SetDebug("fs", true)
+		l.SetDebug("walkfs", true)
+		return
+	}
+	l.SetDebug("fs", strings.Contains(os.Getenv("STTRACE"), "fs"))
```

Comment:
Sorry to be annoying, but this is still too much and removes the `all` variant from `fs`. How about:

```go
sttrace := os.Getenv("STTRACE")
l.SetDebug("fs", strings.Contains(sttrace, "fs") || sttrace == "all") // also matches walkfs
l.SetDebug("walkfs", strings.Contains(sttrace, "walkfs") || sttrace == "all")
```

Reply:
How so? For "all" both "fs" and "walkfs" are set. For "walkfs" additionally "fs" is set, because only setting "walkfs" does not have any effect.

Reply:
In your current code, in the `all` case, you set both to true in the `if`. Then you set `fs` to false after the `if` because the all check is missing there. Not repeating it is cleaner.

Reply:
Honestly, I am (still) a bit confused, as I return from the body of the first `if`, so it implicitly behind a check. I changed it to a `switch`, is this cleaner?